lib/core/src/irods_random.cpp

```diff
@@ -1,7 +1,34 @@
+#include "irods/irods_random.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/rodsErrorTable.h"
+
 #include <boost/random.hpp>
+#include <fmt/format.h>
 #include <openssl/rand.h>
-#include "irods/irods_random.hpp"
+
 #include <ctime>
+#include <random>
+#include <string>
+
+namespace
+{
+    auto generate_random_alphanumeric_character() -> char
+    {
+        constexpr char min_char = '0';
+        constexpr char max_char = 'z';
+
+        static std::random_device rd;
+        static std::default_random_engine e{rd()};
+        static std::uniform_int_distribution<> d{min_char, max_char};
```

Comment:
I wonder if we should pass this into the function rather than make it static?

Passing it in gives us the ability to change the RNG, but I'm not sure if it's worth the effort.

Reply:
I don't think it is worth the effort for 4.3.1. Willing to do it in the future, though. No interfaces would really be changing here since this is in the anonymous namespace. Thoughts?

Reply:
That sounds fine. Certainly not required for 4.3.1.